cat.plugin.ts

```diff
@@ -0,0 +1,85 @@
+import Constants from '../../common/constants';
+import { Plugin } from '../../common/plugin';
+import { ChannelType, IContainer, IHttpResponse, IMessage } from '../../common/types';
+
+class Breed {
+    public name: string = '';
+    public id: string = '';
+}
+
+export class CatPlugin extends Plugin {
+    public name: string = 'Cat Plugin';
+    public description: string = 'Generates pictures of cats.';
+    public usage: string = 'cat <breed (optional)>';
+    public permission: ChannelType = ChannelType.Public;
+    public pluginChannelName: string = Constants.Channels.Public.Pets;
+
+    private _API_URL: string = 'https://api.thecatapi.com/v1/';
+    private _breeds: Breed[] = [];
+
+    constructor(public container: IContainer) {
+        super();
+        //creates list of breeds
+        this.container.httpService
+            .get(`${this._API_URL}breeds`)
+            .then((response: IHttpResponse) => {
+                const breeds = response.data;
+
+                this._breeds = breeds.map((breedData: { name: string, id: string }) => {
+                    return {
+                        name: breedData.name.toLowerCase(),
+                        id: breedData.id.toLowerCase(),
+                    }
+                })
+            })
+            .catch(console.log);
+    }
+
+    public async execute(message: IMessage, args?: string[]) {
+        if (args != undefined && (args[0] === "breeds" || args[0] === "breed")) {
+            //Simply return the list of suported breeds
+            let reply = "";
+
+            this._breeds.forEach(breed => {
+                reply = reply.concat(breed.name, "\n");
+            });
+
+            message.reply(`Breeds supported: \n\`\`\`\n${reply}\`\`\``);
+            return;
+
+        }
+
+        const breedIn = this._parseCommand(args || []);
+
+        let searchCom = "";
+
+        // checks if their was a bread was a breed, then if that breed is recognised
+        if (breedIn.length != 0) {
+
+            const breedEntry = this._breeds.find((breed) => breed.name === breedIn);
+
+            if (breedEntry != undefined) {
+                searchCom = breedEntry.id;
+            } else if (breedIn != "ran") {
```

Comment:
i lack context on this. why would `breedIn` be equal `ran`?

Reply:
Not gonna lie took me a few minutes to remember, it was for a random picture.  I have now renamed it to random.